meteor/server/api/ingest/rundownInput.ts

```diff
@@ -1494,6 +1509,48 @@ function afterIngestChangedData(
 	triggerUpdateTimelineAfterIngestData(rundown.playlistId)
 }
 
+export function handleUpdatedSegmentRanks(
+	peripheralDevice: PeripheralDevice,
+	rundownExternalId: string,
+	segmentExternalIds: string[],
+	ranks: number[]
+) {
+	if (segmentExternalIds.length !== ranks.length) {
+		throw new Meteor.Error(400, `Number of segmentExternalIds does not match number of ranks`)
+	}
+
+	const studio = getStudioFromDevice(peripheralDevice)
+	const rundownId = getRundownId(studio, rundownExternalId)
+	const playlistId = getRundown(rundownId, rundownExternalId).playlistId
+
+	return rundownPlaylistSyncFunction(
+		playlistId,
+		RundownSyncFunctionPriority.INGEST,
+		'handleUpdatedSegmentRanks',
+		() => {
+			const rundown = getRundown(rundownId, rundownExternalId)
+			const playlist = getRundownPlaylist(rundown)
+			const cache = waitForPromise(initCacheForRundownPlaylist(playlist))
+
+			for (const [index, externalId] of segmentExternalIds.entries()) {
+				cache.Segments.update(
```

Comment:
I wonder if there should be a check for the segments existing, and to log a warning if they do not?
I think the update call returns a changed count that can be used for this

Reply:
I have added a check to see if the number of changes is 0 after each update operation, if so then a warning is logged with the segmentExternalId and rundownExternalId similar to other ingest warnings